geonode/social/views.py

```diff
@@ -42,3 +42,26 @@ def get_context_data(self, *args, **kwargs):
             public=True,
             action_object_content_type__model='comment')[:15]
         return context
+
+
+class UserActivity(ListView):
+    """
+    Returns recent user activity.
+    """
+    context_object_name = 'action_list'
+    template_name = 'actstream/actor.html'
+
+    def get_queryset(self):
+        # There's no generic foreign key for 'actor', so can't filter directly
+        # Hence the code below is essentially applying the filter afterwards
+        user_actions = []
+        action_list = Action.objects.filter(public=True)[:15]
+        for action in action_list:
+            if action.actor.username == self.kwargs['actor']:
+                user_actions.append(action)
+        return user_actions
```

Comment:
Not sure if there is a better way to do this since `actor` can't be filtered by. I'm open to feedback and suggestions.